### PR TITLE
Define Super When Undefined

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Set `data_attributes` to `[]` when `super` is undefined.
* Add spec to make sure an exception isn't thrown.

[Finishes: #114719395]

### CODE REVIEW

File: spec/models/content_providers/evaluate_thoughts_provider_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+module ContentProviders
+  describe EvaluateThoughtsProvider, type: :model do
+    describe "when SharedItemAssociated is prepended" do
```
Comment (ericcf) [inline spec/models/content_providers/evaluate_thoughts_provider_spec.rb]: Isn't the module always prepended?

File: spec/models/content_providers/evaluate_thoughts_provider_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      describe "#data_attributes" do
+        describe "when `super` is undefined" do
```
Comment (ericcf) [inline spec/models/content_providers/evaluate_thoughts_provider_spec.rb]: Isn't super always `undefined`?

File: spec/models/content_providers/evaluate_thoughts_provider_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+          it "sets an empty array" do
```
Comment (ericcf) [inline spec/models/content_providers/evaluate_thoughts_provider_spec.rb]: The array is not empty.

**Comment by ericcf** [comment]
@michaelwehrley It just seems like the spec descriptions refer to the implementation rather than behavior. I think they can be simplified. Otherwise looks good!

File: spec/support/model_helpers.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+RSpec.shared_examples "it has shared attributes" do
+  it "returns an array of attributes" do
+    expect(subject.data_attributes)
```
Comment (ericcf) [inline spec/support/model_helpers.rb]: What about `.to include(shared_items_attributes: [:is_public, :action_type])`?

Comment (michaelwehrley) [inline spec/support/model_helpers.rb]: I like. :rocket:

**Comment by ericcf** [comment]
:+1: :100:

**Pull request merged**